klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,766 @@
 
-logger.formatter = proc do |severity, datetime, progname, msg|
-  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10,40].strip + ") : #{msg}
+logger.formatter = proc do |_severity, datetime, _progname, msg|
+  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10, 40].strip + ") : #{msg}
 "
 end
 
 #================================================
 #----------------- FILE SETUP -------------------
 #================================================
-logger.info("Starting running GF180MCU Klayout DRC runset on %s" % [$input])
-logger.info("Ruby Version for klayout: %s" % [RUBY_VERSION])
+logger.info(format('Starting running GF180MCU Klayout DRC runset on %s', $input))
+logger.info(format('Ruby Version for klayout: %s', RUBY_VERSION))
 
 if $input
-    if $topcell
-        source($input, $topcell)
-    else
-        source($input)
-    end
-end
-
-if $table_name
-  table_name = $table_name
-else
-  table_name = "main"
+  if $topcell
+    source($input, $topcell)
+  else
+    source($input)
+  end
 end
 
+table_name = $table_name || 'main'
 
-logger.info("Loading database to memory is complete.")
+logger.info('Loading database to memory is complete.')
 
 if $report
-    logger.info("GF180MCU Klayout DRC runset output at: %s" % [$report])
-    report("DRC Run Report at", $report)
+  logger.info(format('GF180MCU Klayout DRC runset output at: %s', $report))
+  report('DRC Run Report at', $report)
 else
-    layout_dir = Pathname.new(RBA::CellView::active.filename).parent.realpath
-    report_path = layout_dir.join("gf180_drc.lyrdb").to_s
-    logger.info("GF180MCU Klayout DRC runset output at default location: %s" % report_path)
-    report("DRC Run Report at", report_path)
+  layout_dir = Pathname.new(RBA::CellView.active.filename).parent.realpath
+  report_path = layout_dir.join('gf180_drc.lyrdb').to_s
+  logger.info('GF180MCU Klayout DRC runset output at default location: %s', report_path)
+  report('DRC Run Report at', report_path)
 end
 
 #================================================
 #------------------ SWITCHES --------------------
 #================================================
-logger.info("Evaluate switches.")
+logger.info('Evaluate switches.')
 
 # connectivity rules
-if $conn_drc == "true"
+if $conn_drc == 'true'
   CONNECTIVITY_RULES = $conn_drc
-  logger.info("connectivity rules are enabled.")
+  logger.info('connectivity rules are enabled.')
 else
   CONNECTIVITY_RULES = false
-  logger.info("connectivity rules are disabled.")
-end # connectivity rules
+  logger.info('connectivity rules are disabled.')
+end
 
 # WEDGE
-if $wedge == "false"
-  WEDGE = $wedge
-else
-  WEDGE = "true"
-end # WEDGE
+WEDGE = if $wedge == 'false'
+          $wedge
+        else
+          'true'
+        end
 
-logger.info("Wedge enabled  %s" % [WEDGE])
+logger.info(format('Wedge enabled  %s', WEDGE))
 
 # BALL
-if $ball == "false"
-  BALL = $ball
-else
-  BALL = "true"
-end # BALL
+BALL = if $ball == 'false'
```

Comment:
`$ball ||= true` ?

Reply:
Yes

Reply:
Can we keep the original structure of the code then and keep any config refactoring for #37?

Reply:
Same as above

Reply:
Are you intentionally keeping this a `string` rather that converting it to a boolean? I would assume that the rest of the rules would prefer to test this directly rather than doing string comparison?

Reply:
Done, `bool check` function is used for string to boolean conversion.
![image](https://user-images.githubusercontent.com/91015308/218083375-9cfd64cd-e228-4e99-bb45-3ac53b862fff.png)